server/handlers/get_account_test.go

```diff
@@ -35,32 +36,62 @@ func TestGetAccount(t *testing.T) {
 		account, err := app.AccountStore.Create("unlocked@test.com", []byte("bar"))
 		require.NoError(t, err)
 
+		err = app.AccountStore.AddOauthAccount(account.ID, "test", "ID1", "email", "TOKEN1")
+		require.NoError(t, err)
+
+		err = app.AccountStore.AddOauthAccount(account.ID, "trial", "ID2", "email", "TOKEN2")
+		require.NoError(t, err)
+
+		oauthAccounts, err := app.AccountStore.GetOauthAccounts(account.ID)
+		require.NoError(t, err)
+
 		res, err := client.Get(fmt.Sprintf("/accounts/%v", account.ID))
 		require.NoError(t, err)
 		assert.Equal(t, http.StatusOK, res.StatusCode)
-		assertGetAccountResponse(t, res, account)
+
+		assertGetAccountResponse(t, res, account, oauthAccounts)
 	})
 }
 
-func assertGetAccountResponse(t *testing.T, res *http.Response, acc *models.Account) {
+func assertGetAccountResponse(t *testing.T, res *http.Response, acc *models.Account, oAccs []*models.OauthAccount) {
 	// check that the response contains the expected json
-	assert.Equal(t, []string{"application/json"}, res.Header["Content-Type"])
-	responseData := struct {
-		ID                int    `json:"id"`
-		Username          string `json:"username"`
-		LastLoginAt       string `json:"last_login_at"`
-		PasswordChangedAt string `json:"password_changed_at"`
-		Locked            bool   `json:"locked"`
-		Deleted           bool   `json:"deleted_at"`
-	}{}
+	type response struct {
+		ID                int                      `json:"id"`
+		Username          string                   `json:"username"`
+		OauthAccounts     []map[string]interface{} `json:"oauth_accounts"`
+		LastLoginAt       string                   `json:"last_login_at"`
+		PasswordChangedAt string                   `json:"password_changed_at"`
+		Locked            bool                     `json:"locked"`
+		Deleted           bool                     `json:"deleted_at"`
+	}
+
+	var responseData response
 	err := test.ExtractResult(res, &responseData)
 	assert.NoError(t, err)
 
-	assert.Equal(t, acc.Username, responseData.Username)
-	assert.Equal(t, acc.ID, responseData.ID)
-	// NOTE: acc.LastLoginAt is empty so the API returns an empty response
-	assert.Equal(t, "", responseData.LastLoginAt)
-	assert.Equal(t, acc.PasswordChangedAt.Format("2006-01-02T15:04:05Z07:00"), responseData.PasswordChangedAt)
-	assert.Equal(t, false, responseData.Locked)
-	assert.Equal(t, false, responseData.Deleted)
+	sort.Slice(oAccs, func(i, j int) bool {
+		return oAccs[i].Provider < oAccs[j].Provider
+	})
+
+	assert.Equal(t, []string{"application/json"}, res.Header["Content-Type"])
+	assert.Equal(t, responseData, response{
+		ID:       acc.ID,
+		Username: acc.Username,
+		OauthAccounts: []map[string]interface{}{
```

Comment:
Its weird to expect two here, same with the hard coded providers.  You could just loop over oAccs and build up the map I think?

Reply:
Don't think it's a good idea, because if an OAuth provider is disabled after a few registrations won't ever be returned again.

Reply:
I don't understand - the oauth accounts passed in already have a provider that is used for sorting a few lines up. Shouldn't that same provider be used to populate the maps?

Don't understand how disabled providers come into play here either.

Reply:
I misread the code, code updated 😄 .